R/add_sample_metadata.R

```diff
@@ -0,0 +1,45 @@
+#' Add data frame with sample metadata to an SCE object
+#'
+#' @param sce SingleCellExperiment object to add metadata too
+#' @param metadata_df A dataframe with metadata to add to the sce.
+#' @param sample_id_column Name of column in `metadata_df` that contains the sample ids.
+#'   The contents of this column should be equivalent to the `sample_id` in `metadata(sce)`
+#'
+#' @return SingleCellExperiment object with a `metadata_df` added to the `sample_metadata`
+#'   slot of the metadata list
+#' @export
+#'
+add_sample_metadata <- function(sce,
+                                metadata_df,
+                                sample_id_column){
```

Comment:
I made this comment elsewhere, but maybe we should just require that there be a column be named `sample_id`? I think it is just as easy for the user to make that change outside the function, and maybe more sustainable for downstream applications where we might want those columns to match.